### ISSUE TITLE
Rule Request: Preprocessor flag statements inequality check

### ISSUE THREAD

**Issue by kacper1703** [realm/SwiftLint#3369]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

For code that is controlled via preprocessor flags (`#if DEBUG` etc.), an error/warning should appear if all statements embedded in flag check are equal.

**1. Why should this rule be added?** 

Such a check can prevent unexpected runtime behavior if the app doesn't have unit and/or UI tests implemented.

**2. Provide several examples of what _would_ and _wouldn't_ trigger violations.**

Triggering examples:
```swift
static let shouldShowDebugMenu: Bool = {
#if DEBUG
  return true
#else
  return true ↓
#endif
}
```
```swift
var environment: String {
#if DEBUG
  return "Debug"
#elseif BETA
  return "Debug" ↓
#endif
}
```
```swift
var isPrerelease: Bool {
#if DEBUG || BETA
  return true
#else
  return true ↓
#endif
}
```
```swift
let viewController: UIViewController
#if MOCKING
viewController = FooViewController()
#else
viewController = FooViewController() ↓
#endif 
```

Non-triggering examples:
```swift
static let shouldShowDebugMenu: Bool = {
#if DEBUG
  return true
#else
  return false
#endif
}
```
```swift
var environment: String {
#if DEBUG
  return "Debug"
#elseif BETA
  return "Beta"
#endif
}
```
```swift
var isPrerelease: Bool {
#if DEBUG || BETA
  return true
#else
  return false
#endif
}
```
```swift
let viewController: UIViewController
#if MOCKING
viewController = FooViewController()
#else
viewController = BarViewController()
#endif 
```

**3. Should the rule be configurable, if so what parameters should be configurable?**

I can't imagine any options, apart from warning/error level.

**4. Should the rule be opt-in or enabled by default? Why?**

Preprocessor flag checks are hard to catch unless there are unit or UI tests in the project running on different environments. I would say this should be enabled by default since it prevents runtime issues, potentially affecting production environments.

**Comment by jpsim** [comment]
Good idea! Have you seen this often in the wild?

**Comment by agarmash** [comment]
I can take that!
Also, based on the examples, it can be generalized to checking if there is the same piece of code for different cases in general conditional statement

**Comment by kacper1703** [comment]
> Based on the examples, can it be generalized to checking if there is the same piece of code for different cases in a regular conditional statement?

I think that would also be great in case of `if/else/else if` statements:
```swift
if condition {
  return true
} else {
  return true↓
}
```

But in that case handling `switch`es that repeat same code in different cases would also be good:
```swift
switch self.environment {
  case .staging:
    print("Staging")
  case .production:
    print("Staging")↓
}
```

**Comment by kacper1703** [comment]
> Good idea! Have you seen this often in the wild?

Actually it happened to me, we had a debug menu with raw user profile data (like IDs and keys). The menu item was visible on production app due to insufficient code review (snippet below) but luckily it didn't open the menu after tapping 😅
```swift
static let shouldShowDebugMenu: Bool = {
#if DEBUG
  return true
#else
  return true↓
#endif
}()
```
